lib/mix/lib/mix/scm/git.ex

```diff
@@ -155,16 +156,25 @@ defmodule Mix.SCM.Git do
     if branch = opts[:branch] do
       "origin/#{branch}"
     else
-      opts[:ref] || opts[:tag] || "origin/master"
+      opts[:ref] || opts[:tag]
     end
   end
 
-  defp get_rev_info do
-    destructure [origin, rev],
-      :os.cmd('git --git-dir=.git config remote.origin.url && git --git-dir=.git rev-parse --verify --quiet HEAD')
+  defp get_rev_info(opts) do
+    cmd = 'git --git-dir=.git config remote.origin.url && git --git-dir=.git rev-parse --verify --quiet HEAD'
```

Comment:
Actually, I think it should be this:

    defp get_rev_info(rev) do
      cmd = 'git --git-dir=.git config remote.origin.url && git --git-dir=.git rev-parse --verify --quiet #{rev}~0'
      # same as today
    end

Then, in get_lock, we will call:

    get_rev_info("HEAD")

but when checking the lock:

    get_rev_info(opts[:tag] || opts[:branch] || opts[:rev] || "HEAD")

This way we don't need to add this line up above:

    old_rev_as_of_opts?(rev_info, lock_rev)

Because the following line will always do the proper check:

    lock_rev  != rev_info[:rev] 

Does this make sense?